components/Search.tsx

```diff
@@ -0,0 +1,36 @@
+import Router from "next/router";
+import React from "react";
+import styles from '../styles/Search.module.css'
+
+interface SubmitProps {
+  query: string;
+}
+
+interface SearchProps {
+  onSubmit: (props: SubmitProps) => void;
+}
+
+const SearchForm = ({onSubmit}:SearchProps) => {
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    const {query} = event.currentTarget
+    
+    onSubmit({
+      query: query.value,
+    })
+  }
+
+  const handleClick = (e) => {

```

Comment:
This line has no type, but line 14 has a type. Better to choose between `e` and `event` then be consistent with which to use.